### ISSUE TITLE
Submit as elm package

### ISSUE THREAD

**Issue by bitterjug** [adam-r-kowalski/elm-style#2]
This has got to be better then [the other elm-style](http://package.elm-lang.org/packages/poying/elm-style/1.0.0/) ?

**Comment by seanhess** [comment]
I'm looking for something like this. Will you publish to packages.elm-lang.org? If you don't want to I'd be happy to create a fork and do it.

**Comment by adam-r-kowalski** [comment]
Yeah sure you can fork it if you would like, the one issue is that if you upload it as a package, and do not clone this specific repo you wont be able to take advantage of the gulp file, so you won't get automatic reloading of your files.

I would also take a look at [elm-css](https://github.com/rtfeldman/elm-css) because it seems to be really active, and I unfortunately do not have to much time to work on this project.  It might be better to have the community standardize on one css library and focus on making that one as good as it can be.

**Comment by altaic** [comment]
Unfortunately, elm-css is missing a lot of stuff, e.g. gradients. I'd love to see this one on [package.elm-lang.org](http://package.elm-lang.org).

**Comment by seanhess** [comment]
I'll fork and post this week. This package is too useful.

**Comment by seanhess** [comment]
Published: http://package.elm-lang.org/packages/seanhess/elm-style/1.0.0/ - https://github.com/seanhess/elm-style

Feel free to send pull requests for anything missing!

I think we should probably start moving away from having the properties as strings, and instead either represent them as types, or just let people pass in strings for the more complicated ones.

**Comment by seanhess** [comment]
Oh, and I changed a few things: I got rid of the idea of it being a boilerplate with automatic reload, etc. It's only the library of css combinators. Updated documentation, etc.

**Comment by adam-r-kowalski** [comment]
Awesome work @seanhess! I'm glad you took the time to do this.

I have been doing some work with ClojureScript and have been playing around with inline styles there as well. There are a couple of things I have done there that you might want to think about adding to this project in the future.

###### Media Queries

I would pass a tuple of (media queries, styles) to my components which would then behind the scenes use [match media](http://www.sitepoint.com/javascript-media-queries/) to see which queries match and conditionally apply the styles as needed.  Be careful to merge the media query styles with the original styles rather then overriding them.  match media also allows you to create a listener that can inform you when these queries start/stop matching so you can react accordingly.

###### Pseudo Selectors

It seems you may already have a solution for this with [elm-dynamic-style](http://package.elm-lang.org/packages/garetht/elm-dynamic-style/1.0.3/) however, if you want to know what I did I can give a high level overview.  

Since I had macros in Clojure/Script this was simpler so I don't know exactly how you could do it here.  Essentially, since you construct the dom using simple data structures, you can parse through it and attach on-mouse-enter, on-mouse-leave functions which will cause a state change and merge the hover styles with your styles.  Again this gets tricky because if there is already an on-mouse-enter defined you would need to essentially wrap it in a new function which would both handle the state change as well as any pre existing logic in the original function.

**Comment by seanhess** [comment]
Thanks Adam! I'll definitely take a look at those and think about where to
go from here.

**Comment by altaic** [comment]
Wow, great work and thanks for the fast response!
